src/Components/Modal/Modal.tsx

```diff
@@ -1,69 +1,66 @@
-import React from "react";
+import React, { useState } from "react";
 import { observer } from "mobx-react";
 import { Button, Container, Col, Row } from "react-bootstrap";
-import styles from "./Modal.module.css";
+import { ErrorToast } from "../ErrorToast/ErrorToast";
 import type { MediaPresenter } from "../MediaPresenter";
-
+import styles from "./Modal.module.css";
+import classnames from "classnames";
 type Props = {
   mediaPresenter: MediaPresenter;
 };
 
 export const Modal: React.FC<Props> = observer(({ mediaPresenter }) => {
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
   const openModal = (): void => {
-    var modal: HTMLElement | null = document.getElementById("magicModal");
-    if (modal != null) {
-      modal.style.display = "block";
-    }
+    setModalOpen(true);
   };
   const closeModal = (): void => {
     var modal: HTMLElement | null = document.getElementById("magicModal");
     if (modal != null) {
-      modal.style.display = "none";
+      // modal.style.display = "none";
+      setModalOpen(false);
     }
   };
   const windowCloseModal = (
     event: React.MouseEvent<HTMLDivElement, MouseEvent>
   ): void => {
     var modal: HTMLElement | null = document.getElementById("magicModal");
     if (event.target === modal && modal != null) {
-      modal.style.display = "none";
+      // modal.style.display = "none";
+      setModalOpen(false);
     }
   };
   return (
     <div>
       <Button
         variant="info"
-        className={styles.createDesignBtn}
         onClick={openModal}
+        className={styles.createDesignBtn}
       >
         Create a design
       </Button>
 
       <div
         id="magicModal"
```

Comment:
can remove this once the other references are removed

Reply:
I still need this to check if clicked area is not the modal area (line 28 ` if event.target === modal` )

Reply:
Is this div the bg overlay div?
A better approach would be to `stopPropagation` on the modal container, which means that the click event wouldn't be passed through to the background, and call `close`